File: Java-OOP-Basics/Problems Course Instance 2016_2017/src/Lesson01DefiningClasses/Lab/bank_account_advanced/BankAccount.java
package Lesson01DefiningClasses.Lab.bank_account_advanced;

class BankAccount {
    private static int bankAccountCount;

    private int id;
    private double balance;

    BankAccount() {
        this.id = ++bankAccountCount;
    }

    void deposit(double amount) {
        this.balance += amount;
    }

    int getId() {
        return id;
    }

    double getBalance() {
        return balance;
    }
}

File: C++ Programming/Problems/04.InheritanceAdvanced/BankTask/BankAccount.cpp
#include "BankAccount.h"

BankAccount::BankAccount() : BankAccount(0)
{
}

BankAccount::BankAccount(long double balance) : _balance(balance)
{
}

long double BankAccount::getBalance()
{
	return this->_balance;
}

void BankAccount::setBalance(long double money)
{
	this->_balance = money;
}
